### PR TITLE
Add standard deviation quantizer

### LINKED ISSUE #43: Fancier fitting methods than min/max
MXNet has other means of fitting quantization parameters:

http://on-demand.gputechconf.com/gtc/2017/presentation/s7310-8-bit-inference-with-tensorrt.pdf

https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/contrib/quantization.py#L315

https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/contrib/quantization.py#L338

### PR DESCRIPTION
**Pull request by XapaJIaMnu**
This should fix #43 
All tests pass. Please double check that my reference slow implementation is correct:
https://github.com/kpu/intgemm/blob/stdQuantizer/test/quantize_test.cc#L33

**CONTENT UNRELATED TO PULL REQUEST:**
Bonus question: Why do those lines not work? I tried it in an independent test file and it seems to work just fine.
https://github.com/kpu/intgemm/blob/stdQuantizer/test/quantize_test.cc#L36
https://github.com/kpu/intgemm/blob/stdQuantizer/test/quantize_test.cc#L37

Here's the independent test case:
```cpp
#include <vector>
#include <iostream>
#include <cmath>
#include <algorithm>

template <class integral>
float QuantMultStdRef(std::vector<integral>& vals, int stdnum) {
  int num_items = vals.size();
  float normal_sums = 0;
  float squares_sum = 0;
  std::for_each(vals.begin(), vals.end(), [&] (int n) {normal_sums+=n;});
  std::for_each(vals.begin(), vals.end(), [&] (int n) {squares_sum+=n*n;});

  float mean = normal_sums/num_items;
  return mean + (std::sqrt((squares_sum/num_items) - (mean*mean)) * stdnum);;
}

int main() {
    std::vector<int> a = {1,2,3,4,5,6,7,8,9,10};
    float stQuantMultStdRef = QuantMultStdRef(a, 1);
    std::cout << "QuantMultStdRef: " << stQuantMultStdRef << std::endl;
}
```

### CODE REVIEW

**Comment by mateuszchudyk** [comment]
It looks like AlignedVector doesn't want to cooperate with std::for_each. I'll check that.

**Comment by mateuszchudyk** [comment]
Just a thought. What do you think about splitting GetQuantizerStd into two functions: GetMatrixStats that returns a pair (mean, stddev) and GetQuantizerStd which just use that function. I'm just wondering if we can use information about mean and stddev somewhere else in the future.

**Comment by kpu** [comment]
I'm in agreement that statistics calculation is separately useful.

**Comment by kpu** [comment]
Also maybe this is overkill but we have multiple horizontal reduction implementations and it would be nice to factor out sum and max (used in MaxAbsolute) in 32-bit int and float.  Though our abstraction layer with overloaded max_ps isn't good at turning into a template argument because we actually have 3 max_ps functions, one for each width.

File: intgemm.cc [hunk 1]
```diff
@@ -32,6 +32,8 @@ const CPUType kCPU = ChooseCPU(CPUType::AVX512VNNI, CPUType::AVX512BW, CPUType::
 
 float (*MaxAbsolute)(const float *begin, const float *end) = ChooseCPU(avx512f::MaxAbsolute, avx512f::MaxAbsolute, avx2::MaxAbsolute, sse2::MaxAbsolute, sse2::MaxAbsolute, Unsupported_MaxAbsolute);
 
+float (*GetQuantizerStd)(const float *begin, const float *end, int stdnum) = ChooseCPU(AVX512VNNI_8bit::GetQuantizerStd, AVX512_8bit::GetQuantizerStd, AVX2_8bit::GetQuantizerStd, SSSE3_8bit::GetQuantizerStd, Unsupported_8bit::GetQuantizerStd, Unsupported_8bit::GetQuantizerStd);
```
Comment (kpu) [inline intgemm.cc]: Why should the number of standard deviations be an integer?

Comment (kpu) [reply intgemm.cc]: Also, why not just provide mean and stddev?

**Comment by XapaJIaMnu** [comment]
How about now?

**Pull request merged**